### ISSUE TITLE
[Feature Request] Goyo mode?

### ISSUE THREAD

**Issue by shaunsingh** [kdav5758/TrueZen.nvim#18]
Hi. First of all I love your extension and the work you do on it!

By any chance could you implement a goyo mode as well? E.g. running :Goyo will also add padding to the top and bottom. I love using Ataraxis mode for programming but for writing pose goyo looks and feel a lot better, but I would rather have one extension over two

**Comment by kdav5758** [comment]
Hey! Goyo and TrueZen are two different things, if you want Goyo to have top and bottom padding as well, consider opening an issue in their repo.

As for TrueZen, how could I improve your experience with the plugin? You said something about "the writing pose being better in Goyo"... could you explain to me that a little bit more? (if possible, provide a screenshot)

Oh an also:
* remember that there were new features and settings introduced to Ataraxis mode two days ago, so try to tweak them around to see if you can get your ideal look and feel.
* As a personal recommendation, go and grab [comfortable-motion.vim](https://github.com/yuttie/comfortable-motion.vim) to make the scrolling smoother!

**Comment by shaunsingh** [comment]
Sorry for not phrasing the question properly 

Goyo has more of a "square" look to it: 
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196912/117051233-4e3be080-ace4-11eb-81e3-a7a5cf1674b8.png">

Compared to ataraxis mode, which aligns more with the intellij-style zen mode: 

![Uploading image.png…]()

My question meant to be, is there a way to create a command to launch ataraxis mode with custom arguments, so that ataraxis mode retains its default functionality but I can have a "square" look when needed? p.s., thank you for showing me comfortable motion, its a very nice plugin

**Comment by kdav5758** [comment]
Could you reupload the last image... It seems you accidentally commented before the image loaded:

![image](https://user-images.githubusercontent.com/58336662/117058930-dc17cb80-ace4-11eb-97de-cf90759affb8.png)

**Comment by shaunsingh** [comment]
sorry about that, the internet is acting up today 

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/71196912/117059371-bfcc5c80-aced-11eb-9ea7-0c55618b163d.png">

**Comment by kdav5758** [comment]
Oh that's quite simple then! You can get this config to get that boxy look:

```lua
	ataraxis = {
		ideal_writing_area_width = 0,
		just_do_it_for_me = false,
		left_padding = 30,
		right_padding = 30,
		top_padding = 3,
		bottom_padding = 3,
	}

```

With boxy look:
![image](https://user-images.githubusercontent.com/58336662/117061181-8133a380-ace7-11eb-9d91-da811ec519c2.png)

Without boxy look:
![image](https://user-images.githubusercontent.com/58336662/117061404-c1932180-ace7-11eb-9910-d7fa82e52068.png)

Is this what you want?

**Comment by shaunsingh** [comment]
Pretty much, but I was wondering If I could put it as a command? For example in Goyo you can do something like this 

`:Goyo 50%+25%x50%-25%`

where you can pass parameters to the goyo command, by any chance is there something/could you add something similar to ataraxis or focus mode?

**Comment by kdav5758** [comment]
Done! in this case I didn't add percentages but rather `vim cells` as the measurement so that it's easier to understand. Be sure to balance them or else it might not look like you wanted it to.

You can check check it from the Notices section: [https://github.com/kdav5758/TrueZen.nvim#-notices](https://github.com/kdav5758/TrueZen.nvim#-notices)
And you can learn how to use it visiting the Default Commands sectio: [https://github.com/kdav5758/TrueZen.nvim#default](https://github.com/kdav5758/TrueZen.nvim#default)

In short, you can do this with:
```
:TZAtaraxis l10 r10 t3 b1
```
Where `l` = left, `r` = right, `t` = top, and `b` = bottom.

**Comment by shaunsingh** [comment]
Thank you so much!

**Comment by kdav5758** [comment]
Wow... that was fast haha. You're welcome! just keep in mind that `vim cells` work differently from what you were used to in Goyo. Their upside is that they are more accurate, easier to implement and easier to understand.

Enjoy :)

**Comment by shaunsingh** [comment]
Yup, it works even better than Goyo. I will, thank you once again

**Comment by shaunsingh** [comment]
Sorry to open the issue again, but the status line is a bit finicky. If I run the Ataraxis command with some parameters, (e.g. :TZAtaraxis l45 r45 t2 b2`) the status line still stays (I'm using lualine at the moment)

[image](https://user-images.githubusercontent.com/71196912/117378314-22b32480-aea3-11eb-844e-55e021d7450d.png)

Additionally, not sure if its related, but when I exit it looks a tad bit messed up: 

![image](https://user-images.githubusercontent.com/71196912/117378409-57bf7700-aea3-11eb-9d77-b18715e317ee.png)

**Comment by kdav5758** [comment]
Hey! don' worry about reopening the issue, I'l still help you if you need help!

So, first of all:

1. You statusline dissapears or it's not behaving as expected

Sadly every statusline works kind of differently so the way I, for instance, hid Galaxyline, is not the same way I hide Lualine. That's why I created the integrations. For statuslines without integrations people can open issues with commands or snippets or something that one can use to hide *x* statusline.

TrueZen will try its best to make your current statusline behave, if that fails, you should use the `force_hide_statusline` setting to **forcefully** make the statusline **behave**. I was looking at your **vimrc** repo and in your `init.vim` this setting was not enabled:

![image](https://user-images.githubusercontent.com/58336662/117388484-7415e080-aeb0-11eb-9560-d103004f4ff6.png)

2. E86: Buffer 28 does not exist

Yes, this issue ***existed*** and it was actually [the first issue opened](https://github.com/kdav5758/TrueZen.nvim/issues/1) in the repo. This was caused by the user not having `splitbelow` and `splitright` set in their configs. However, there is no need to enable this settings in your config (although I personally recommend you doing), because I already solved this with some magic and in the src code you actually find were the fix is located at:

![image](https://user-images.githubusercontent.com/58336662/117388814-facabd80-aeb0-11eb-8b43-e45aa2d187b6.png)

So yeah, no sure what's going on. Could you update to latest commit at main, configure the settings properly for your use case and then try to see if you get this issue again? If you do get the issue, could you explain to me how to replicate it?

**Comment by shaunsingh** [comment]
Hi! Setting force_hide_statusline worked perfectly, the status line seems to behave now, however, I'm still getting the second error.

If I run :TZAtaraxis by itself, it seems to work fine. However, running it with parameters at the end is what causes the issue to show up. If you run something like `:TZAtaraxis l45 r45 t2 b2`, it should show up 

The issue persists even after setting split below and split right, so I assume something else is the issue

**Comment by kdav5758** [comment]
Could you describe step by step how to reproduce it?

**Comment by shaunsingh** [comment]
1. load up nvim with truezen
2. Run `:TZAtaraxis l45 r45 t2 b2`
3. Run `:TZAtaraxis l45 r45 t2 b2` again to exit

**Comment by kdav5758** [comment]
![image](https://user-images.githubusercontent.com/58336662/117390207-56964600-aeb3-11eb-82bc-c2a03bedb7c0.png)

In the README I specified to NOT exit Ataraxis giving `:TZAtaraxis` with params or else stuff might break. Could you try again but without params? so:

1. load up nvim with truezen
2. Run :TZAtaraxis l45 r45 t2 b2
3. Run :TZAtaraxis l45 r45 t2 b2 again to exit

**Comment by shaunsingh** [comment]
I apologize for the mistake. However the issue persists even when exiting with :TZAtaraxis


https://user-images.githubusercontent.com/71196912/117390420-23a48000-aebc-11eb-98f0-e1bca25ed675.mov

**Comment by kdav5758** [comment]
Wait but didn't you just use `:TZAtaraxis l45 r45 t2 b2` again to exit?

**Comment by shaunsingh** [comment]
No, I use <leader>z which should be binded to :TZAtaraxis 
<img width="480" alt="image" src="https://user-images.githubusercontent.com/71196912/117390639-88f87100-aebc-11eb-9324-28743df8b142.png">

The gui I'm using is called neovide! Its a bit buggy at times but its absolutely beutiful

**Comment by kdav5758** [comment]
Are you sure it' not because of NeoVide that it's not working? I have never used it before and I don't know what things are different there... Could you try your config in vanilla nvim nightly? I because I just tested your config and it was working as expected:


https://user-images.githubusercontent.com/58336662/117390915-96a9f880-aeb4-11eb-85fe-ae2f323529d0.mp4

Of course, I didn't install all of the plugin you had and all, but it's working!

**Comment by shaunsingh** [comment]
Doesn't seem to be working on terminal nvim either, one of my plugins must be interfering with it. I'll comment then out and get back to you on the issue

**Comment by kdav5758** [comment]
Okay I already commented out everything but the settings and TrueZen's config but I'm still not able to reproduce the issue.

**Comment by kdav5758** [comment]
Finally I was able to reproduce it!!
![image](https://user-images.githubusercontent.com/58336662/117391458-978f5a00-aeb5-11eb-93c6-6cda3fec6e46.png)

Indeed there is something with your config, I'll remove stuff until it works and update you when I find the culprit.

**Comment by kdav5758** [comment]
Hey I found the issue! My apologies for assuming it was an issue with your config. Setting these two values to zero causes the issue:
![image](https://user-images.githubusercontent.com/58336662/117393089-f73b3480-aeb8-11eb-8839-31bdbb5c7206.png)

I was also able to replicate it in my own config. For now set both to one and it'll work. I'm sorry for the inconvenience, I'll fix it now.

**Comment by shaunsingh** [comment]
Thank you for all your hard work on the project. Works like a charm

**Comment by kdav5758** [comment]
Done. The fix is ready and it's working as expected. You can now set those two values to zero if you want to and it'll work as expected.

![image](https://user-images.githubusercontent.com/58336662/117396186-10df7a80-aebf-11eb-8ee8-9ebe4f31afd2.png)

I'll merge the bug fix commits to the `main` branch now. Thanks for pointing this issue out!

**Comment by shaunsingh** [comment]
Works perfectly, thank you! Btw, running `:TZAtaraxis l45 r45 t2 b2` to exit as well works perfectly fine. Not sure if something changed but thought I should let you know. Thanks once again for all the help
